examples/pi/pi.cc

```diff
@@ -0,0 +1,72 @@
+//===------------------------------------------------------------*- C++ -*-===//
+//
+//                                     SHAD
+//
+//      The Scalable High-performance Algorithms and Data Structure Library
+//
+//===----------------------------------------------------------------------===//
+//
+// Copyright 2018 Battelle Memorial Institute
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy
+// of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+//===----------------------------------------------------------------------===/
+
+#include <iostream>
+#include <iomanip>
+#include <random>
+
+#include "shad/core/array.h"
+#include "shad/core/algorithm.h"
+#include "shad/core/numeric.h"
+#include "shad/core/execution.h"
+#include "shad/data_structures/one_per_locality.h"
+
+namespace shad {
+
+int main(int argc, char *argv[]) {
+
+  shad::array<size_t, 128> counters;
+
+  size_t numberOfPoints = 1e12;
+  size_t numberOfPointsPerSim = numberOfPoints / counters.size();
+
+  shad::generate(
```

Comment:
I would prefer a `transform` over an array of seeds, so that the output does not change with the execution policy.

Reply:
Do you want me to change it in a transform?